Source/PeleLM.cpp

```diff
@@ -7412,42 +7449,83 @@ PeleLM::mac_sync ()
+        const Real be_cn_theta_SDC = 0.5;
         const int visc_coef_comp = Temp;
 
-        // Diffuse the sync.  On exit,
-        //      Snp1[0][temp] = rho^{postsync} * delT
-        //      SpecDiffusionFluxnp1[RhoH] = - lambda.Grad(delT)
-        diffusion->diffuse_scalar(Sn, Sn, Snp1, Snp1, Temp, 1, Density,
-                                  prev_time,curr_time,be_cn_theta_SDC,RhT,rho_flagT,
-                                  SpecDiffusionFluxn,SpecDiffusionFluxnp1,nspecies+2,
-                                  &Trhs,0,&RhoCp_post,0,
-                                  betan,betanp1,nspecies,visc_coef,visc_coef_comp,
-                                  crse_ratio,theBCs[Temp],geom,
-                                  solve_mode,add_old_time_divFlux,diffuse_this_comp);
-
-        deltaT_iter_norm = Snp1[0]->norm0(Temp);
+        LPInfo info;
+        info.setAgglomeration(1);
+        info.setConsolidation(1);
+        info.setMetricTerm(false);
+        MLABecLaplacian deltaTSyncOp({geom}, {grids}, {dmap}, info);
+        deltaTSyncOp.setMaxOrder(diffusion->maxOrder());
+        deltaTSyncOp.setScalars(1.0, dt*be_cn_theta_SDC);
+        deltaTSyncOp.setACoeffs(0.0, RhoCp_post);
+
+        AMREX_D_TERM(MultiFab lambdax(*betanp1[0], amrex::make_alias, nspecies, 1);,
+                     MultiFab lambday(*betanp1[1], amrex::make_alias, nspecies, 1);,
+                     MultiFab lambdaz(*betanp1[2], amrex::make_alias, nspecies, 1););
+        std::array<MultiFab*,AMREX_SPACEDIM> bcoeffs{AMREX_D_DECL(&lambdax,&lambday,&lambdaz)};
+        deltaTSyncOp.setBCoeffs(0.0,amrex::GetArrOfConstPtrs(bcoeffs));
+
+        std::array<LinOpBCType,AMREX_SPACEDIM> mlmg_lobc;
+        std::array<LinOpBCType,AMREX_SPACEDIM> mlmg_hibc;
+        const Vector<BCRec>& theBCs = AmrLevel::desc_lst[State_Type].getBCs();
+        const BCRec& bc = theBCs[Temp];
+        Diffusion::setDomainBC(mlmg_lobc, mlmg_hibc, bc); // Same for all comps, by assumption
+        deltaTSyncOp.setDomainBC(mlmg_lobc, mlmg_hibc);
+
+        MLMG deltaTSyncSolve(deltaTSyncOp);
+
+        const Real S_tol     = visc_tol;
+        const Real S_tol_abs = visc_tol * Trhs.norm0(0);
+
+        deltaTSyncSolve.solve({&deltaT}, {&Trhs}, S_tol, S_tol_abs); 
+
+
+
+        //// Diffuse the sync.  On exit,
+        ////      Snp1[0][temp] = delT
+        ////      SpecDiffusionFluxnp1[RhoH] = - lambda.Grad(delT)
+        //diffusion->diffuse_scalar(Sn, Sn, Snp1, Snp1, Temp, 1, Density,
+        //                          prev_time,curr_time,be_cn_theta_SDC,RhT,rho_flagT,
+        //                          SpecDiffusionFluxn,SpecDiffusionFluxnp1,nspecies+2,
+        //                          &Trhs,0,&RhoCp_post,0,
+        //                          betan,betanp1,nspecies,visc_coef,visc_coef_comp,
+        //                          crse_ratio,theBCs[Temp],geom,
+        //                          solve_mode,add_old_time_divFlux,diffuse_this_comp);
+
```

Comment:
Can we stop the practice of leaving old versions of the code in place by commented.  The file is huge enough already, and nobody is ever going to want to look at the broken version...especially when it is not marked as broken.  The next person coming along will wonder what this code is and will not be able to use it or understand why it is there.

Reply:
This has been removed indeed ... I don't know why it pops-up there.

Reply:
I think you are looking at an outdated commit.

Reply:
weird, something happened in my windows too. the commit shows as outdated now and it didn't 5 minutes ago.